### ISSUE TITLE
Windows 10 - Running pinned taskbar shortcut creates a second taskbar entry

### ISSUE THREAD

**Issue by ryancabanas** [dbeaver/dbeaver#3070]
Hi,

I'm running DBeaver 4.3.5 on Windows 10.  If I make a shortcut to the "dbeaver.exe" file and pin this shortcut to my taskbar, if I click on this shortcut in the taskbar, a 2nd icon is created in the taskbar to run the app.

![image](https://user-images.githubusercontent.com/27093222/36923661-f5cb5a5c-1e39-11e8-9b02-6fe0a314ff84.png)

**Comment by serge-rider** [comment]
Can't reproduce.
When I pin DBeaver to taskbar I see another icon (brown beaver in a circle: 
![image](https://user-images.githubusercontent.com/6398845/36931788-9f12a586-1ece-11e8-9e8e-575fce497b62.png) )
When I run it it turns into white beaver head.

How do you run dbeaver in the first place - clicking on desktop icon, from command line, from Windows menu?

**Comment by ryancabanas** [comment]
Hi Serge,

Thanks for looking into this.

![image](https://user-images.githubusercontent.com/27093222/36937395-7ebd9fd6-1ee0-11e8-96b1-c5b7266b1512.png)

![image](https://user-images.githubusercontent.com/27093222/36937414-ddaaadae-1ee0-11e8-9c0d-666f252dda30.png)

![image](https://user-images.githubusercontent.com/27093222/36937418-0de467c6-1ee1-11e8-90d8-e8d35cf4985c.png)

![image](https://user-images.githubusercontent.com/27093222/36937444-53e21a02-1ee1-11e8-86c6-bb666234e186.png)

![image](https://user-images.githubusercontent.com/27093222/36937499-5fa57f22-1ee2-11e8-9a4c-d99ff44bc4b1.png)

![image](https://user-images.githubusercontent.com/27093222/36937581-7d0fdd40-1ee3-11e8-9e98-0d6eb7ad984d.png)


Please let me know if you have any other questions.  Thanks!

**Comment by serge-rider** [comment]
Yep, it is clear about icon (indeed we use different icons for different sizes, forgot about that).

Regarding pinning to taskbar. I've been able to reproduce this issue using your steps to reproduce.
Now I know yet another way to pin something to Windows taskbar :) I'll investigate this. For some reason Windows don't recognize dbeaver process when it is run that way.

Workaround (or "the right way") - just run DBeaver (from menu or explorer) and then, after it will load, right click on it in taskbar and choose "Pin to taskbar".

**Comment by ryancabanas** [comment]
Hi Serge,

Thanks again.  I'm using the ZIP archive file method.

![image](https://user-images.githubusercontent.com/27093222/36939751-ae5d08b6-1f04-11e8-8645-d76bb2f4037e.png)

![image](https://user-images.githubusercontent.com/27093222/36939768-edc2c5ae-1f04-11e8-8c31-855069e62b4c.png)

Thanks!

**Comment by kohanyirobert** [comment]
I've encountered the same problem. This isn't a DBeaver thing, but an Eclipse thing. To resolve it one needs to configure the `dbeaver.ini` file and add an extra flag like this before the `-vmargs` flag.

```
-vm
c:/Program Files/Java/jdk-9.0.1/bin/server/jvm.dll
```

Obviously needs to be amended for the actual Java version installed on the machine. After this it works. I'm not sure if this problem appears when using an installer, but it happens when using the ZIP distribution. Hope this helps.

**Comment by ryancabanas** [comment]
Hi @kohanyirobert .  Thanks!  That worked!  This is probably nothing that can be changed with the ZIP distribution, since each machine's "jvm.dll" path will differ, correct?  If not, then I will just close this this item.  Thanks again.

**Comment by kohanyirobert** [comment]
I think the ZIP distro can't do anything about it, you can't specify a dynamic path to jvm.dll, so yeah I guess you can close this and maybe mention this issue somewhere in the docs, not sure where.

**Comment by ryancabanas** [comment]
Okay.  I'll document it as soon as I can and then close this item with a reference.  Thanks.

**Comment by nmat** [comment]
This happened for me also before.

However I think it was when I pinned the startup icon to the taskbar that it then created a second icon on launch the next time. Not 100% sure it is related though.

**Comment by jah-mann** [comment]
Hello,

No big deal to me, but this also happens to me. I downloaded using the Windows 64-bit installer, not the ZIP archive. I'm on Windows 10. I do not have Java installed, although my jvm.dll file can be found here:
C:\Program Files\DBeaver\jre\bin\server\jvm.dll

I tried inserting the flag in dbeaver.ini, right before -vmargs:
-vm
jre/bin/server/jvm.dll 

Also tried doing the full path to the file starting with C:

But in both cases, when I re-pin dbeaver.exe to the taskbar and launch the application, a duplicate icon is still created.

**Comment by jah-mann** [comment]
I've found a workaround for this, for me. Hopefully this works for others encountering problems... 

I simply unpinned from taskbar, launched the application first (dbeaver.exe, not the shortcut), and then re-pinned to taskbar while the application was already open. I can then exit and re-launch the application, and it no longer duplicates the icon in the taskbar.

**Comment by p91paul** [comment]
Yes that's the trick. It's very important not to pin during splash screen, but only when the application is fully opened. That's been an eclipse bug since forever...

**Comment by kenjib2** [comment]
You can also reproduce this by starting DBeaver, and then pinning it quickly before the program finishes loading. It seems like there are two different processes -- a loading process and then the actual running application process. If the loading process gets pinned, Windows does not register that the actual app is the one that is pinned. That's how it looks to me...

If you just wait for the app to fully load it works fine.

**Comment by stotes** [comment]
Even when the app was pinned after fully loading I could not get the icon to pin with an icon.

Here was my solution.

1. PS `get-appxpackage > apps.txt`
2. Find Dbeaver get the PackageFamilyName in the txt that matches with the Target Type you from the properties of the app startmenu shortcut.
3. Create a shortcut putting the `PackageFamilyName` into the target: 
`C:\Windows\explorer.exe shell:appsFolder\{{PackageFamilyName}}!Dbeaver`
4. Change Icon from the dbeaver.exe

In my case the shortcut looks like this:
`C:\Windows\explorer.exe shell:appsFolder\DBeaverCorp.DBeaverCE_1b7tdvn0p0f9y!DBeaver`

**Comment by ghost** [comment]
The same issue is happening to me in Windows 7. 

A workaround I found was a combination of all the workarounds presented here.

For me, when the application was fully launched (after the splashscreen) there was no "Pin to taskbar" option when I right-clicked it. To fix that, first I had to create a shortcut of `dbeaver.exe` then added the following arguments to the shortcut's "Target": `-vm "C:\Program Files\Java\jdk-18.0.2\bin\server\jvm.dll"` (you have to change the path to your actual Java path).

Now, when I launch the application through that shortcut and wait for the splashscreen to go away, I have a "Pin to taskbar" option. If you pin that one, then it doesn't duplicate when launching the app.
